src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -146,27 +160,22 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                 }
             }
 
-            private void CreateInitialSolutionChecksum()
+            private async Task SynchronizeAssets()
             {
-                // initial solution checksum creation won't be affected by global operation.
-                // cancellation can only happen if it is being shutdown.
-                Task.Run(async () =>
-                {
-                    await UpdateSolutionChecksumAsync(ShutdownCancellationToken).ConfigureAwait(false);
+                _synchronize = false;
 
-                    var remoteHostClient = await _service.GetRemoteHostClientAsync(ShutdownCancellationToken).ConfigureAwait(false);
-                    if (remoteHostClient == null)
-                    {
-                        return;
-                    }
+                var remoteHostClient = await _service.GetRemoteHostClientAsync(ShutdownCancellationToken).ConfigureAwait(false);
+                if (remoteHostClient == null)
```

Comment:
@ManishJayaswal @srivatsn What's the current POR around OOP?  The last time we discussed this, i thought the idea was that we wanted this to operate like a critical service that features could depend on (similar to things like the Threadpool) and whatnot.  Has that plan changed?  If so, what's the current thinking and what's the estimate on costs around such plans? 

Thanks!